Automattic/wp-calypso
# Add checkbox to indicate external user

#### Changes proposed in this Pull Request

* Adds checkbox when administrator, editor, author, or contributor role is selected to indicate that a user is an external user in invite.
<img width="450" alt="Screen Shot 2019-05-21 at 4 39 17 PM" src="https://user-images.githubusercontent.com/2810519/58137448-22062280-7be7-11e9-8d4f-40e82f768fbc.png">
<img width="450" alt="Screen Shot 2019-05-23 at 9 08 16 AM" src="https://user-images.githubusercontent.com/2810519/58268483-5e966300-7d3a-11e9-9234-10ac0008f347.png">
* Add indicator to external user when viewing people list
<img width="450" alt="Screen Shot 2019-05-21 at 4 39 02 PM" src="https://user-images.githubusercontent.com/2810519/58137441-1dda0500-7be7-11e9-977e-448059ff96e5.png">

* Add checkbox when administrator, editor, author, or contributor role is selected to indicate that a user is an external user in edit user form
<img width="450" alt="Screen Shot 2019-05-21 at 4 39 08 PM" src="https://user-images.githubusercontent.com/2810519/58137444-203c5f00-7be7-11e9-9dfe-660736d639c6.png">


#### Testing instructions

1. ~Note:  testing this PR requires D28186-code~ D28186 has been merged

1.  Invite another test user to a blog as an administrator, editor, author, or contributor and external user via `/people/new/:siteSlug`
1. Accept the above invite
1. Return to primary test user and confirm that in `/people/team/:siteSlug` the new user has the `Contractor` badge: <img width="300" alt="Screen Shot 2019-05-21 at 4 44 59 PM" src="https://user-images.githubusercontent.com/2810519/58137661-e029ac00-7be7-11e9-9f96-e6647a7f60f3.png">
1. Click on the user and toggle the "This user is a contractor, freelancer, consultant, or agency" checkbox, and click "Save Settings"
1. Navigate back to `/people/team/:siteSlug` and confirm the contractor badge is gone
1. Return to the edit screen for the invited user and re click the checkbox, confirming that the user regains their contract badge
1. Delete the user via the button on the bottom on the same page
1. Return to `/people/team/:siteSlug` and reload with dev tools open
1. Find the `external-contributors` request in the network tab and confirm the body is an empty array
